playwright/_impl/_browser_context.py

```diff
@@ -298,8 +299,7 @@ async def _pause(self) -> None:
     async def storage_state(self, path: Union[str, Path] = None) -> StorageState:
         result = await self._channel.send_return_as_dict("storageState")
         if path:
-            with open(path, "w") as f:
-                json.dump(result, f)
+            await async_writefile(path, "w", json.dumps(result))
```

Comment:
I was already leaving this comment, should be `async_write_file(page, json.dumps(result))`

Reply:
Okay but how would you write in binary with this ?

Reply:
OR you mean we should have two functions for writing one for str and one for bytes ?

Reply:
`async_write_file` would accept `Union[str, bytes]` and will pick the right mode.

Reply:
Sure! Thanks for confirming